src/core/qgsmaphittest.cpp

```diff
@@ -205,3 +206,17 @@ void QgsMapHitTest::runHitTestLayer( QgsVectorLayer *vl, SymbolSet &usedSymbols,
   }
 }
 
+bool QgsMapHitTest::rasterVisible( QgsRasterLayer *layer ) const
+{
+  if ( ! layer->dataProvider() )
+    return false;
+  QgsRectangle footprint = layer->dataProvider()->extent();
+  if ( mSettings.destinationCrs() != layer->crs() )
+  {
+    QgsCoordinateTransform ct = QgsCoordinateTransform( mSettings.destinationCrs(), layer->crs(), mSettings.transformContext() );
```

Comment:
Shouldn't source be the `layer->crs()` and target the `mSettings.destinationCrs()`?
Your test could handle a case of crs transformation to be sure.